index.html

```diff
@@ -2013,51 +1971,94 @@ <h2>Content Metadata</h2>
-        the value of the DID Resolution Metadata <strong>error</strong> property MUST be <strong>representationNotSupported</strong> as
-        defined in section <a href="#did-resolution-metadata"></a>.
-    </p>
-	<h2>methodNotSupported</h2>
-	<p> If a DID method is not supported during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>methodNotSupported</strong>.
+	<p>
+		The algorithms described in this specification throw specific types of errors.
+		Implementers might find it useful to convey these errors to other libraries or
+		software systems. This section provides specific URLs and descriptions for the errors,
+		such that an ecosystem implementing technologies described
+		by this specification might interoperate more effectively when errors occur. Additionally,
+		this specification uses some errors defined in <a data-cite="CID#processing-errors">Section 
+		3.5 Processing Errors</a> of the [[CID]] specification.
+	</p>
+		
+	<p>
+		Implementers SHOULD use
+		[[RFC9457]] to encode the error data structure. If [[RFC9457]] is used:
 	</p>
-	<h2>internalError</h2>
-	<p>When an unexpected error occurs during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>internalError</strong>.
-    </p>
-	<h2>invalidPublicKey</h2>
-	<p>If an invalid public key value is detected during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>invalidPublicKey</strong>.
-    </p>
-	<h2>invalidPublicKeyLength</h2>
-	<p>If the byte length of <i>rawPublicKeyBytes</i> does not match the expected public key length for the associated multicodecValue during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>invalidPublicKeyLength</strong>.
-    </p>
-	<h2>invalidPublicKeyType</h2>
-	<p>If an invalid public key type is detected during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>invalidPublicKeyType</strong>.
-    </p>
-	<h2>unsupportedPublicKeyType</h2>
-	<p>If an unsupported public key type is detected during <a>DID Resolution</a> or <a>DID URL dereferencing</a>,
-		the value of the DID Resolution or DID URL Dereferencing Metadata <strong>error</strong> property MUST be <strong>unsupportedPublicKeyType</strong>.
-    </p>
+
+	<ul>
+		<li>
+The `type` value of the error object MUST be a URL that starts with the value
+`https://w3id.org/security#` and ends with the value in the section listed
```

Comment:
Hmm, we might want to put DID-specific error codes in the DID vocabulary?

Reply:
I updated the PR to try to address this point